src/lib/autoinstall/autosetup_helpers.rb

```diff
@@ -159,7 +152,28 @@ def autosetup_network
     # @return [Boolean] true when network config should be written before the
     #   proposal; false when not
     def network_before_proposal?
-      !!@network_before_proposal
+      return @network_before_proposal unless @network_before_proposal.nil?
+
+      @network_before_proposal = profile_section("networking").fetch("setup_before_proposal", false)
+    end
+
+    # Convenience method to fetch the content of the given section name from
+    # the current {Yast::Profile}
+    #
+    # @param name [String] section name to be fetched from the profile
+    # @return [Hash] the profile section with the given name when present;
+    #   an empty hash when not
+    def profile_section(name)
+      Yast::Profile.current[name] || {}
```

Comment:
Why an empty hash? The partitioning section, for instance, should be an array. Perhaps it is the only exception ((I do not know). In this case, and having into account the calls you do to this method, using a hash is safe enough. But, please, document this behaviour.

Reply:
Basically the convenience method was added because a pattern detected in the code:

https://github.com/yast/yast-autoinstallation/pull/620/files#diff-6aba7d6785a22cfa0b01b3cbb9251771R454
https://github.com/yast/yast-autoinstallation/pull/606/files#diff-0100ef44fa49d992794bb1d6b470d681L169
https://github.com/yast/yast-autoinstallation/pull/620/files#diff-6aba7d6785a22cfa0b01b3cbb9251771R454

And for networking, was not there but used same approach:
  https://github.com/yast/yast-autoinstallation/pull/620/files#diff-6aba7d6785a22cfa0b01b3cbb9251771R454

If there are exceptions, then probably is better to not generalize it as then do not see any benefit at all.

Reply:
I have added a `@note`, but if you prefer I can remove it ad bring back the `general_section method`.

Reply:
Finally removed the convenience methods.